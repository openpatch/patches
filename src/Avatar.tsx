/** @jsx jsx */
import { jsx, css } from "@emotion/core";
import { CSSProperties, Fragment } from "react";
import { width } from "styled-system";
import { Box, BoxProps } from "./Box";
import { useBreakpoints } from "./hooks";
import { Image, ImageProps } from "./Image";
import { ResponsiveProp } from "./system/types";

export type AvatarProps = {
  size?: ResponsiveProp<"small" | "medium" | "large" | string>;
  username?: string;
  src?: string;
  placeholder?: boolean;
};

const getSize = (size: "small" | "medium" | "large" | string | undefined) => {
  switch (size) {
    case "small":
      return "30px";
    case "medium":
      return "60px";
    case "large":
      return "90px";
    default:
      return size;
  }
};

export const Avatar = ({
  src,
  username,
  size = ["medium"],
  placeholder = false,
}: AvatarProps) => {
  if (!Array.isArray(size)) {
    size = [size];
  }

  const responsiveSize = useBreakpoints(size);
  const width = getSize(responsiveSize);

  return (
    <Fragment>
      {!placeholder ? (
        <Image
          borderRadius="full"
          src={src}
          width={width}
          height={width}
          alt={`${username} avatar`}
          title={`${username} avatar`}
        />
      ) : (
        <Box
          as="span"
          display="inline-block"
          borderRadius="full"
          width={width}
          height={width}
          borderStyle="solid"
          borderColor="neutral.300"
        />
      )}
    </Fragment>
  );
};
