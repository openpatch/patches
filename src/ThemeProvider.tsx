import React, { ReactElement } from "react";
import { ThemeProvider as EThemeProvider } from "emotion-theming";
import { Theme, base } from "./themes";
import { GlobalCssTheme } from "./themes/utils";

interface ThemeProviderProps {
  theme?: Theme;
  children: ReactElement;
}

const ThemeProvider = ({ theme = base, children }: ThemeProviderProps) => {
  return (
    <EThemeProvider theme={theme}>
      <GlobalCssTheme theme={theme} />
      {children}
    </EThemeProvider>
  );
};

export default ThemeProvider;
