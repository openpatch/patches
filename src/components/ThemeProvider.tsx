import React, { ReactElement } from "react";
import {
  ThemeProvider as SCThemeProvider,
  DefaultTheme,
} from "styled-components";
import defaultTheme from "../shared/theme";

interface ThemeProviderProps {
  theme?: DefaultTheme;
  children: ReactElement;
}

const ThemeProvider = ({
  theme = defaultTheme,
  children,
}: ThemeProviderProps) => {
  return <SCThemeProvider theme={theme}>{children}</SCThemeProvider>;
};

export default ThemeProvider;
