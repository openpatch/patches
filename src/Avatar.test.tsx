import { render } from "@testing-library/react";
import { Avatar, AvatarProps } from "./Avatar";
import { ThemeProvider } from "./ThemeProvider";

const renderWithTheme = (props?: AvatarProps) =>
  render(
    <ThemeProvider>
      <Avatar {...props} />
    </ThemeProvider>
  );

test("should render", () => {
  renderWithTheme();
});

test("should match snapshot", () => {
  const { container } = renderWithTheme();
  expect(container).toMatchSnapshot();
});
