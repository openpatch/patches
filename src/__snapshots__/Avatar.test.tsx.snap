// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.emotion-0 {
  border-radius: 9999px;
}

<div>
  <div
    id="patches-root"
  >
    <img
      class="emotion-0 emotion-1"
      height="60px"
      width="60px"
    />
  </div>
</div>
`;
